napari/conftest.py

```diff
@@ -292,6 +293,125 @@ def tmp_plugin(npe2pm_: TestPluginManager):
+    return ViewerModel()
+
+
+@pytest.fixture
+def qt_viewer_(qtbot, viewer_model, monkeypatch):
+    from napari._qt.qt_viewer import QtViewer
+
+    viewer = QtViewer(viewer_model)
+
+    original_controls = viewer.__class__.controls.fget
+    original_layers = viewer.__class__.layers.fget
+    original_layer_buttons = viewer.__class__.layerButtons.fget
+    original_viewer_buttons = viewer.__class__.viewerButtons.fget
+    original_dock_layer_list = viewer.__class__.dockLayerList.fget
+    original_dock_layer_controls = viewer.__class__.dockLayerControls.fget
+    original_dock_console = viewer.__class__.dockConsole.fget
+    original_dock_performance = viewer.__class__.dockPerformance.fget
+
+    def hide_widget(widget):
+        widget.hide()
+
+    def hide_and_clear_qt_viewer(viewer: QtViewer):
+        viewer._instances.clear()
+        viewer.hide()
+
+    def patched_controls(self):
+        if self._controls is None:
+            self._controls = original_controls(self)
+            qtbot.addWidget(self._controls, before_close_func=hide_widget)
+        return self._controls
+
+    def patched_layers(self):
+        if self._layers is None:
+            self._layers = original_layers(self)
+            qtbot.addWidget(self._layers, before_close_func=hide_widget)
+        return self._layers
+
+    def patched_layer_buttons(self):
+        if self._layersButtons is None:
+            self._layersButtons = original_layer_buttons(self)
+            qtbot.addWidget(self._layersButtons, before_close_func=hide_widget)
+        return self._layersButtons
+
+    def patched_viewer_buttons(self):
+        if self._viewerButtons is None:
+            self._viewerButtons = original_viewer_buttons(self)
+            qtbot.addWidget(self._viewerButtons, before_close_func=hide_widget)
+        return self._viewerButtons
+
+    def patched_dock_layer_list(self):
+        if self._dockLayerList is None:
+            self._dockLayerList = original_dock_layer_list(self)
+            qtbot.addWidget(self._dockLayerList, before_close_func=hide_widget)
+        return self._dockLayerList
+
+    def patched_dock_layer_controls(self):
+        if self._dockLayerControls is None:
+            self._dockLayerControls = original_dock_layer_controls(self)
+            qtbot.addWidget(
+                self._dockLayerControls, before_close_func=hide_widget
+            )
+        return self._dockLayerControls
+
+    def patched_dock_console(self):
+        if self._dockConsole is None:
+            self._dockConsole = original_dock_console(self)
+            qtbot.addWidget(self._dockConsole, before_close_func=hide_widget)
+        return self._dockConsole
+
+    def patched_dock_performance(self):
+        if self._dockPerformance is None:
+            self._dockPerformance = original_dock_performance(self)
+            qtbot.addWidget(
+                self._dockPerformance, before_close_func=hide_widget
+            )
+        return self._dockPerformance
+
```

Comment:
This lack of coverage is expected as I `monkeypatch` all possible properties, that may spawn qt object. 

Should I mark it with `# pragma: no cover`?